pep_sphinx_extensions/pep_theme/templates/not_found.html

```diff
@@ -0,0 +1,44 @@
+{# 404 Not Found: Redirect typos and link to PEP 0 #}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="color-scheme" content="light dark" />
+    <title>PEP Not Found | peps.python.org</title>
+    <link rel="shortcut icon" href="{{ pathto('_static/py.png', resource=True) }}"/>
+    <link rel="stylesheet" href="{{ pathto('_static/style.css', resource=True) }}" type="text/css" />
+    <link rel="stylesheet" href="{{ pathto('_static/mq.css', resource=True) }}" type="text/css" />
+    <link rel="stylesheet" href="{{ pathto('_static/dark.css', resource=True) }}" type="text/css" media="(prefers-color-scheme: dark)" id="css-dark"/>
+    <link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,700;1,400&display=swap" rel="stylesheet">
+    <meta name="description" content="Python Enhancement Proposals (PEPs)"/>
+    <script>
+    const pepNumMatch = window.location.pathname.match(/^\d+$/)
```

Comment:
With `^` and `$` in the regex, it won't match `pep-NNN`.  Did you just change this temporarily to test?
Note that in the suggestion below I used `match[0]` because of this change, but if my previous suggestion is restored it should be `match[1]`.

Reply:
>it won't match `pep-NNN`

From above, "I don't want to include the `pep-` prefix for the auto-redirects"

A

Reply:
Why not?
If we set the autolinking to `https://peps.python.org/pep-<num>/` and num is a valid 4-digits number (e.g. pep-0008), then it will go straight to the right page.  If we omit the `pep-` prefix, then it will have to go through an extra redirect if JS is enabled, or end up at the 404 page if it's disabled.

In case of a valid PEP with a <4 digits number (e.g. pep-8) it will go to `/pep-8/` and get redirected to `/pep-0008`.
In case of an invalid PEP with a <4 digits number (e.g. pep-123) it will go to `/pep-123/`, get redirected to `/pep-0123` and then stop at the 404 page since the PEP doesn't exist and the id is already 4 digits long.

Am I missing something?

Reply:
The autolinking on this repo is currently set to link to `https://python.org/dev/peps/pep-<num>`.  My suggestion that simply matched `pep-(\d+)` works even with this type of URL, but fails with `/^\d+$/`.